site/content/integrate/apps/interactivity/dynamic-form-example/_index.md

```diff
@@ -0,0 +1,544 @@
+
+What if the list of available tags is too large to be computed at once, and we need to use an autocomplete API to retrieve these values dynamically? We will use a `dynamic_select` field for this.
+
+Let's change the field's `type` to `dynamic_select`, and remove the `options` property:
+
+```ts
+const tagsField: AppField = {
+    name: 'tags',
+    modal_label: 'Tags',
+    type: 'dynamic_select',
+    value: [],
+    multiselect: true,
+};
+```
+
+This will use the form's `call` to fetch these values at the endpoint `/forms/create-ticket/lookup`. The same `lookup` path is used for any `dynamic_select` field of this form. We know what field is requesting the autocomplete by accessing `call.selected_field`. We can get the user's query typed in by accessing `call.query`:
+
+```ts
+app.post('/forms/create-ticket/lookup', async (req, res) => {
+    const call: AppCallRequest = req.body;
+
+    // For this form, we only have one dynamic select field, so we should only be receiving requests for that field. This block is here to show that this endpoint could be called for another field, if the form had multiple dynamic select fields, and we would need to handle them individually here when that field is being queried for autocomplete.
+    if (call.selected_field !== 'tags') {
+        const callResponse: AppCallResponse = {
+            type: 'error',
+            error: `Unexpected lookup field: ${call.selected_field}`,
+        };
+
+        res.json(callResponse);
+        return;
+    }
+
+    // TicketTag represents a tag in the external ticket system
+    let tags: TicketTag[];
+    try {
+        // Use some other service to process the autocomplete request
+        tags = await autocompleteTags(call.query);
+    } catch (e) {
+        const callResponse: AppCallResponse = {
+            type: 'error',
+            error: `Error performing autocomplete: ${e.message}`,
+        };
+
+        res.json(callResponse);
+        return;
+    }
+
+    const selectOptions: AppSelectOption[] = tags.map(tag => ({
+        label: tag.name,
+        value: tag.id,
+    }));
+
+    const callResponse: AppCallResponse = {
+        type: 'ok',
+        data: {
+            items: selectOptions,
+        },
+    };
+
+    res.json(callResponse);
+});
+```
+
+Now we have our form integrated with an external system's autocomplete service, with just a few lines of code. Note that we also have access to all of the currently selected values in the form whenever we receive a `lookup` request. For example, we have access to the current Summary and Description through `call.values.summary` and `call.values.description`.
```

Comment:
0/5, but I feel the pieces of code are a big too big and the explanations too short. Maybe you can put the whole code (as it is now) but take a bit more of time dissecting the code or extracting the important parts and going a bit deeper into those. I know you are doing partly that with the comments, but reading code comments feel less natural than reading the actual documentation.